src/shared_modules/indexer_connector/src/monitoring.hpp

```diff
@@ -19,61 +19,122 @@
 constexpr auto INTERVAL = 60u;
 
-namespace HealthCheckRows
-{
-    enum Type : std::size_t
-    {
-        HEADER = 0,
-        DATA,
-        SIZE
-    };
-};
+// 5 seconds timeout for health check requests
+constexpr auto HEALTH_CHECK_TIMEOUT_MS = 5000u;
 
-namespace HealthCheckColumns
-{
-    enum Type : std::size_t
-    {
-        EPOCH = 0,
-        TIMESTAMP,
-        CLUSTER,
-        STATUS,
-        NODE_TOTAL,
-        NODE_DATA,
-        DISCOVERED_CLUSTER_MANAGER,
-        SHARDS,
-        PRI,
-        RELO,
-        INIT,
-        UNASSIGN,
-        PENDING_TASKS,
-        MAX_TASK_WAIT_TIME,
-        ACTIVE_SHARDS_PERCENT,
-        SIZE
-    };
-};
+constexpr auto SERVER_HEALTH_FIELD_NAME {"status"};
 
 /**
  * @brief Monitoring class.
  *
  */
 class Monitoring final
 {
-    std::map<std::string, bool> m_values;
+    std::map<std::string, bool> m_servers;
     std::thread m_thread;
     std::mutex m_mutex;
     std::condition_variable m_condition;
     std::atomic<bool> m_stop {false};
     uint32_t m_interval {INTERVAL};
 
+    /**
+     * @brief Checks the health of a server.
+     *
+     * @note It sends a request to the \p serverAddress and update the serverStatus. The \p authentication object is
+     * used to provide secure communication.
+     *
+     * @param serverAddress Server's address.
+     * @param authentication Object that provides secure communication.
+     */
+    void healthCheck(const std::string& serverAddress, const SecureCommunication& authentication)
+    {
+        auto& serverStatus = m_servers[serverAddress];
+
+        // Set the server status to unavailable by default
+        serverStatus = false;
+
+        // On success callback
+        const auto onSuccess = [&serverStatus](std::string response)
+        {
+            // Parse the response without throwing exceptions
+            // Response example:
+            // [
+            //     {
+            //         "epoch": "1726271464",
+            //         "timestamp": "23:51:04",
+            //         "cluster": "wazuh-cluster",
+            //         "status": "green",
+            //         "node.total": "1",
+            //         "node.data": "1",
+            //         "discovered_cluster_manager": "true",
+            //         "shards": "166",
+            //         "pri": "166",
+            //         "relo": "0",
+            //         "init": "0",
+            //         "unassign": "0",
+            //         "pending_tasks": "0",
+            //         "max_task_wait_time": "-",
+            //         "active_shards_percent": "100.0%"
+            //     }
+            // ]
+
+            const auto data = nlohmann::json::parse(response, nullptr, false).at(0);
+
+            // Check if the server is green
+            if (!data.is_discarded() && data.contains(SERVER_HEALTH_FIELD_NAME) &&
+                data.at(SERVER_HEALTH_FIELD_NAME).get_ref<const std::string&>().compare("green") == 0)
```

Comment:
Based on this documentation: https://www.elastic.co/guide/en/elasticsearch/reference/current/red-yellow-cluster-status.html

Difference Between Yellow and Red Status:

Yellow Status:
	•	Primary shards are allocated, meaning all the data is accessible.
	•	Some replica shards are unassigned, which reduces fault tolerance since there’s no replica of certain data on other nodes.
	•	The data is still readable and writable, but in the event of another node failure, data availability may be compromised due to the lack of replicas.
Red Status:
	•	One or more primary shards are unallocated, meaning some of the data is inaccessible and possibly lost.
	•	Data loss is a real risk in this state, and the cluster can’t fully serve read or write operations for the unallocated primary shards.
	•	The cluster is in a critical state, and immediate action is needed to recover or reallocate the primary shards to restore data availability.

In summary, yellow status means the cluster is operational but lacks fault tolerance due to missing replicas, while red status indicates that some data is unavailable and needs immediate attention.

Reply:
I will update the logic to allow the use of yellow clusters